internal/groups/service.go

```diff
@@ -64,6 +65,18 @@ func (svc service) CreateGroup(ctx context.Context, token, kind string, g groups
 		if err != nil {
 			return groups.Group{}, errors.Wrap(errParentUnAuthz, err)
 		}
+		allowedGrps, err := svc.auth.ListAllObjects(ctx, &magistrala.ListObjectsReq{
+			SubjectType: auth.DomainType,
+			Subject:     res.GetDomainId(),
+			Permission:  auth.DomainRelation,
+			ObjectType:  auth.GroupType,
+		})
```

Comment:
This will be extremely inefficient, especially since we already have domain ID in the group itself. We only need to be able to extract the domain from the request. There are two ways of doing it: updating API to use domain ID or enabling domain extraction from the token (using Auth or direct JWT inspection but without validation). I will get back to you soon with the path forward once the team discusses it internally.

Reply:
@dborovcanin  @arvindh123 Without this PR the you can't add a parent group to a child without them being in the same domain because of ` groupPreConditions` in the `auth/spicedb/policies`. The challenge is that the error logging is inadequate, only showing "failed to add policies" without specifying the cause of the failure.

Reply:
If we add a domain ID as the prefix to this API, we could simply check if the parent group exists and is in the same domain using Repository, rather than calling Auth, right?

Reply:
Yes. We will not need call Auth if we implement it this way.